src/main/java/org/opensearch/security/action/apitokens/ApiTokenRepository.java

```diff
@@ -13,18 +13,59 @@
 
 import java.util.List;
 import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
 
 import com.google.common.annotations.VisibleForTesting;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 
 import org.opensearch.client.Client;
 import org.opensearch.cluster.service.ClusterService;
 import org.opensearch.index.IndexNotFoundException;
 import org.opensearch.security.authtoken.jwt.ExpiringBearerAuthToken;
 import org.opensearch.security.identity.SecurityTokenManager;
+import org.opensearch.security.user.User;
+
+import static org.opensearch.security.http.ApiTokenAuthenticator.API_TOKEN_USER_PREFIX;
 
 public class ApiTokenRepository {
     private final ApiTokenIndexHandler apiTokenIndexHandler;
     private final SecurityTokenManager securityTokenManager;
+    private static final Logger log = LogManager.getLogger(ApiTokenRepository.class);
+
+    private final Map<String, Permissions> jtis = new ConcurrentHashMap<>();
+
+    void reloadApiTokensFromIndex() {
+        Map<String, ApiToken> tokensFromIndex = apiTokenIndexHandler.getTokenMetadatas();
```

Comment:
ApiTokenIndexHandler seems to do a `actionGet`, i.e., a sync  operation. Whenever we're on transport operations, I'd strongly recommend to prefer async operations using action listeners, as otherwise there's a slight deadlock risk due to thread pool exhaustion.